pkg/mcp/testing/reporter.go

```diff
@@ -0,0 +1,59 @@
+package mcptest
+
+import (
+	"google.golang.org/grpc/codes"
+)
+
+type errorCodeKey struct {
+	error string
+	code  codes.Code
+}
+
+type requestKey struct {
+	typeURL      string
+	connectionID int64
+}
+
+type InMemoryStatsContext struct {
```

Comment:
We probably shouldn't assume Reporter methods are called from the same context/goroutine. These should be protected with a mutex.